lib/plugins/audit.js

```diff
@@ -5,52 +5,54 @@
 var assert = require('assert-plus');
 var bunyan = require('bunyan');
 var HttpError = require('restify-errors').HttpError;
+var VError = require('verror');
 
 
 ///--- API
 
+
 /**
- * Returns a Bunyan audit logger suitable to be used in a server.on('after')
- * event.  I.e.:
- *
- * server.on('after', restify.auditLogger({ log: myAuditStream }));
- *
- * This logs at the INFO level.
+ * @param {object} opts The options object.
+ * @param {object} opts.log The logger.
+ * @param {string} opts.event The event from the server which initiates the
+ * log, one of 'pre', 'routed', or 'after'
+ * @param {object} [opts.server] The restify server, used to emit the audit log
+ * object programttically
+ * @param {boolean} [opts.printLog] Whether to print the log via the logger.
  *
  * @public
  * @function auditLogger
- * @param   {Object}   options at least a bunyan logger (log).
- * optionally .server object to emit log. .logBuffer which is a
- * ringbuffer object to store certain amount of log,
- * printLog flag, default is true,
- * if printlog set to false, user need to pass in server object
- * listen to auditlog event to get log information
- * @returns {Function}         to be used in server.after.
+ * @returns {Function}
+ * @emits audit when an audit log has been generated
  */
-function auditLogger(options) {
-    assert.object(options, 'options');
-    assert.object(options.log, 'options.log');
-    assert.optionalObject(options.server, 'options.server');
-    assert.optionalObject(options.logBuffer, 'options.logBuffer');
-    assert.optionalBool(options.printLog, 'options.printLog');
-    //use server object to emit log data
-    var server = options.server;
-    //use logMetrics ringbuffer to store certain period of log records
-    var logMetrics = options.logBuffer;
-    //default always print log
-    var printLog = options.printLog;
+function auditLogger(opts) {
+    assert.object(opts, 'opts');
+    assert.object(opts.log, 'opts.log');
+    assert.string(opts.event, 'opts.event');
+    assert.optionalObject(opts.server, 'opts.server');
+    assert.optionalBool(opts.printLog, 'opts.printLog');
+
+    if (opts.event !== 'after' && opts.event !== 'pre' &&
+        opts.event !== 'routed') {
+        throw new VError('opts.event must be %s, %s, or %s, but is %s', 'pre',
+            'routed', 'after', opts.event);
+    }
+
+    var server = opts.server;
+    var printLog = opts.printLog;
 
     if (typeof printLog === 'undefined') {
         printLog = true;
     }
     var errSerializer = bunyan.stdSerializers.err;
 
-    if (options.log.serializers && options.log.serializers.err) {
-        errSerializer = options.log.serializers.err;
+    if (opts.log.serializers && opts.log.serializers.err) {
+        errSerializer = opts.log.serializers.err;
```

Comment:
I know this was here before, but why does audit logger need serializer? Seems reasonable to expect the parent logger to have already set the serializers it wants? Any customization would happen at `serializers` rather than at `serializers.err`.